contrib/mkimage/debootstrap

```diff
@@ -107,7 +107,7 @@ if [ -d "$rootfsDir/etc/apt/apt.conf.d" ]; then
 	# _keep_ us lean by effectively running "apt-get clean" after every install
 	aptGetClean='"rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true";'
 	echo >&2 "+ cat > '$rootfsDir/etc/apt/apt.conf.d/docker-clean'"
-	cat > "$rootfsDir/etc/apt/apt.conf.d/docker-clean" <<-EOF
+	cat > "$rootfsDir/etc/apt/apt.conf.d/docker-clean" <<-'EOF'
```

Comment:
Does this actually still work with the quotes? Are the `$aptGetClean` variables still expanded?

Reply:
Yes and yes.

The added quotes are only protecting the `EOF` string. Since it is a fixed string, with no variables or special chars, the quotes are really useless here. I'm just adding them for consistency because they are already used that way at lines: 65, 96, 134, 143.

But the optimal solution, in my opinion, would be to remove the useless quotes on those other lines, leaving the bare `<<-EOF`. Do you agree?

Reply:
Indeed, this one does _not_ work with the quotes -- `<<-EOF` vs `<<-'EOF'` is the same as `"string"` vs `'string'`; one applies interpolation, the other does not (which is why this file prefers the quoted form except in this one case where interpolation is required).

Reply:
See https://www.gnu.org/software/bash/manual/bashref.html#Here-Documents.

Reply:
Specifically:

> If any part of _word_ is quoted, the _delimiter_ is the result of quote removal on _word_, and the lines in the here-document are not expanded.

Reply:
To illustrate:
```console
$ cat <<-EOF
$a $b $c
EOF
  
$ cat <<-'EOF'
$a $b $c
EOF
$a $b $c
```

Reply:
Oh, I didn't know about this difference! 😞 

Thank for the explanation and the link. I'll close this PR. Sorry for the noise!

Reply:
No problem, you're one of today's lucky 10,000. :smile:

(https://xkcd.com/1053/)